code/modules/reagents/reagent_containers/food/drinks.dm

```diff
@@ -78,6 +78,39 @@
 	else
 		user << SPAN_NOTICE("\The [src] is full!")
 
+/obj/item/weapon/reagent_containers/food/drinks/Initialize()
+	. = ..()
+	if(is_open_container())
+		verbs += /obj/item/weapon/reagent_containers/food/drinks/proc/gulp_whole
+
+/obj/item/weapon/reagent_containers/food/drinks/proc/gulp_whole()
+	set name = "Gulp Down"
+	set category = "Object"
+	set src in view(1)
+
+	if(is_open_container())
+		if(ishuman(usr))
+			var/mob/living/carbon/human/H = usr
+			if(!H.check_has_mouth())
+				H << "Where do you intend to put \the [src]? You don't have a mouth!"
+				return
+			var/obj/item/blocked = H.check_mouth_coverage()
+			if(blocked)
+				H << SPAN_WARNING("\The [blocked] is in the way!")
+				return
+
+		usr.visible_message(SPAN_NOTICE("[usr] prepares to gulp down [src]."), SPAN_NOTICE("You prepare to gulp down [src]."))
```

Comment:
Some more tweaks needed here;

Only show the prepare message if the swallowing is going to take a while. I'd say 2.5 seconds. So, check if the volume is > 25

We need a visible message describing what happens if its interrupted. Or does standard_splash_mob already trigger one? What happens there

Reply:
It does trigger one. It says something like "blahblah has been splashed by blahblah".